File: LeetCode/0121. Best Time to Buy and Sell Stock/Solution.java
/**
 * 200 / 200 test case passed
 * Status: Accepted
 * Runtime: 1 ms
 */
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) return 0;
        int profit = 0, min = prices[0];
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] < min) {
                min = prices[i];
            } else {
                int diff = prices[i] - min;
                if (diff > profit) profit = diff;
            }
        } 
        return profit;
    }
}

File: LeetCode/0121. Best Time to Buy and Sell Stock/Solution.kt
/**
 * 200 / 200 test cases passed.
 * Status: Accepted
 * Runtime: 224 ms
 */
class Solution {
    fun maxProfit(prices: IntArray): Int {
        if (prices.isEmpty()) return 0
        var profit = 0
        var min = prices[0]
        for (i in 1 until prices.size) {
            if (min > prices[i]) min = prices[i]
            else {
                val diff = prices[i] - min
                profit = if (diff > profit) diff else profit
            }
        }
        return profit
    }
} 
